connector/src/main/java/com/google/cloud/bigquery/connector/common/IteratorMultiplexer.java

```diff
@@ -0,0 +1,137 @@
+package com.google.cloud.bigquery.connector.common;
+
+import java.util.Iterator;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.arrow.util.Preconditions;
```

Comment:
Please change to the guava `Preconditions`, it has the same implementation and will make it easier to re-use the class elsewhere if needed.

Reply:
done.